### ISSUE TITLE
`Factory.attr(_, value)` support value function

### ISSUE THREAD

**Issue by ivawzh** [antonversal/typeorm-factory#12]
Hi. 

I am trying to use Faker with `Factory.attr`. Currently, the second argument is limited to `T[K]` from `attr<K extends keyof T>(name: K, value: T[K])`.

Would you please make it support `value: T[K] | () => T[K]`?

**Comment by RodolfoSilva** [comment]
Hello @antonversal, great job with this library. Thanks for share.
Do you still maintaining it? If you are, I can send you a PR to implement this feature.

**Comment by RodolfoSilva** [comment]
Hello @ivawzh, you can use the `sequence` method to do that:

```ts
.sequence('username', (_) => faker.internet.userName());
```

**Comment by ivawzh** [comment]
Hi @RodolfoSilva, thanks for the workaround tips.

Meanwhile imho, it'd still be better to have an intuitive API that uses the right names for the right things. `sequence` is normally referred to a sequential number-embedded generator. `value` function is better reflecting a stateless attribute value generator.

**Comment by RodolfoSilva** [comment]
I agree, something like https://preview.adonisjs.com/guides/database/factories

**Comment by webbrain-one** [comment]
Hi! 👋 Thanks for the suggestion. I've opened a pull request to address this:

https://github.com/antonversal/typeorm-factory/pull/34

This change updates `Factory.attr` to support passing a function for the value argument, allowing `value: T[K] | () => T[K]`. Please feel free to review and share any feedback. Thanks!
